### PR TITLE
Add enable_rosout into NodeOptions.

### PR DESCRIPTION
**Pull request by Barry-Xu-2018**

### CODE REVIEW

File: rclcpp/include/rclcpp/node_options.hpp [hunk 1]
```diff
@@ -152,6 +152,22 @@ class NodeOptions
...
   use_global_arguments(bool use_global_arguments);
 
+  /// Return the disable_rosout flag.
+  RCLCPP_PUBLIC
+  bool
+  disable_rosout() const;
+
+  /// Set the disable_rosout flag, return this for parameter idiom.
+  /**
+   * If true this will cause the node not to use rosout publisher
```
Comment (fujitatomoya) [inline rclcpp/include/rclcpp/node_options.hpp]:
```suggestion
   * If true this will cause the node not to use rosout logging
```

Comment (Barry-Xu-2018) [reply rclcpp/include/rclcpp/node_options.hpp]: OK.

File: rclcpp/include/rclcpp/node_options.hpp [hunk 2]
```diff
@@ -152,6 +152,22 @@ class NodeOptions
...
   use_global_arguments(bool use_global_arguments);
 
+  /// Return the enable_rosout flag.
+  RCLCPP_PUBLIC
+  bool
+  enable_rosout() const;
+
+  /// Set the enable_rosout flag, return this for parameter idiom.
+  /**
+   * If false this will cause the node not to use rosout logging
```
Comment (hidmic) [inline rclcpp/include/rclcpp/node_options.hpp]: @Barry-Xu-2018 missing trailing period.

File: rclcpp/include/rclcpp/node_options.hpp [hunk 3]
```diff
@@ -152,6 +152,22 @@ class NodeOptions
...
+   *
+   * Defaults to true for now, as there are still some cases where it is not
```
Comment (hidmic) [inline rclcpp/include/rclcpp/node_options.hpp]: @Barry-Xu-2018 this statement is a bit vague, and it reads funny. Did you mean to say _"... where it IS desirable."_?

**Comment by hidmic** [comment]
@Barry-Xu-2018 friendly ping.

File: rclcpp/include/rclcpp/node_options.hpp (hunk 3 above)
Comment (Barry-Xu-2018) [reply rclcpp/include/rclcpp/node_options.hpp]: Yes.  It is my mistake.

File: rclcpp/include/rclcpp/node_options.hpp (hunk 2 above)
Comment (Barry-Xu-2018) [reply rclcpp/include/rclcpp/node_options.hpp]: Yes. I will fix it.

**Comment by Barry-Xu-2018** [comment]
@hidmic 

Thanks for your review.  
I have updated codes based on your comments.  
Please review again.

**Comment by Barry-Xu-2018** [comment]
@hidmic 

> LGTM, though as one last request before I run CI, @Barry-Xu-2018 would you mind extending [`test_node_options.cpp`](https://github.com/ros2/rclcpp/blob/master/rclcpp/test/test_node_options.cpp) to cover the new API? Thanks for your patience!

OK. I have added test codes.  
Please review them.

File: rclcpp/test/test_node_options.cpp [hunk 4]
```diff
@@ -98,3 +98,14 @@ TEST(TestNodeOptions, bad_ros_args) {
...
+TEST(TestNodeOptions, enable_rosout) {
+  auto options = rclcpp::NodeOptions();
+  EXPECT_TRUE(options.enable_rosout());
+
+  options.enable_rosout(false);
+  EXPECT_FALSE(options.enable_rosout());
+
+  options.enable_rosout(true);
+  EXPECT_TRUE(options.enable_rosout());
+}
```
Comment (hidmic) [inline rclcpp/test/test_node_options.cpp]: @Barry-Xu-2018 this is fine, but I think it'd be slightly better to test if after an `options.enable_rosout(boolean_value)` that gets reflected in `options.get_rcl_node_options().enable_rosout`.

File: rclcpp/src/rclcpp/node_options.cpp [hunk 5]
```diff
@@ -198,6 +199,19 @@ NodeOptions::use_global_arguments(bool use_global_arguments)
...
+bool
+NodeOptions::enable_rosout() const
+{
+  return this->enable_rosout_;
+}
+
+NodeOptions &
+NodeOptions::enable_rosout(bool enable_rosout)
+{
+  this->enable_rosout_ = enable_rosout;
```
Comment (hidmic) [inline rclcpp/src/rclcpp/node_options.cpp]: @Barry-Xu-2018 thinking about the test made me realize you probably want to `node_options_.reset()` here.

**Comment by Barry-Xu-2018** [comment]
@hidmic 

Codes has been updated.  
Please review again.

**Comment by hidmic** [comment]
Alright, going in! Thank you for your contribution @Barry-Xu-2018.

**Pull request merged**